src/transaction/eth.ts

```diff
@@ -116,59 +118,144 @@ export const prepareStoreDataTransaction = async (body: CreateRecord, provider?:
 };
 
 /**
- * Sign Ethereum or supported ERC20 transaction with private keys locally. Nothing is broadcast to the blockchain.
+ * Sign Ethereum mint ERC 20 transaction with private keys locally. Nothing is broadcast to the blockchain.
  * @param body content of the transaction to broadcast
  * @param provider url of the Ethereum Server to connect to. If not set, default public server will be used.
  * @returns transaction data to be broadcast to blockchain.
  */
-export const prepareEthOrErc20SignedTransaction = async (body: TransferEthErc20, provider?: string) => {
-    await validateBody(body, TransferEthErc20);
-    const {
-        fromPrivateKey,
-        to,
-        amount,
-        currency,
-        fee,
-        data,
-        nonce,
-        signatureId
-    } = body;
-
-    const client = getClient(provider);
-    client.eth.accounts.wallet.clear();
-    client.eth.accounts.wallet.add(fromPrivateKey);
-    client.eth.defaultAccount = client.eth.accounts.wallet[0].address;
-
-    let tx: TransactionConfig;
-    const gasPrice = fee ? client.utils.toWei(fee.gasPrice, 'gwei') : await ethGetGasPriceInWei();
-    if (currency === Currency.ETH) {
-        tx = {
-            from: 0,
-            to: to.trim(),
-            value: client.utils.toWei(`${amount}`, 'ether'),
-            gasPrice,
-            data: data ? (client.utils.isHex(data) ? client.utils.stringToHex(data) : client.utils.toHex(data)) : undefined,
-            nonce,
-        };
-    } else {
-        // @ts-ignore
-        const contract = new client.eth.Contract([TRANSFER_METHOD_ABI], CONTRACT_ADDRESSES[currency]);
-        const digits = new BigNumber(10).pow(CONTRACT_DECIMALS[currency]);
-        tx = {
-            from: 0,
-            to: CONTRACT_ADDRESSES[currency],
-            data: contract.methods.transfer(to.trim(), `0x${new BigNumber(amount).multipliedBy(digits).toString(16)}`).encodeABI(),
-            gasPrice,
-            nonce,
-        };
-    }
+export const prepareEthMintErc20SignedTransaction = async (body: MintErc20, provider?: string) => {
+  await validateBody(body, MintErc20);
+  const {
+    fromPrivateKey,
+    amount,
+    to,
+    contractAddress,
+    nonce,
+    signatureId,
+  } = body;
+
+  const client = getClient(provider);
+  client.eth.accounts.wallet.clear();
```

Comment:
these 3 lines should not be here - they are in the getClient method

Reply:
this sequence is used everywhere in the file